psdk_wrapper/src/modules/telemetry.cpp

```diff
@@ -1727,6 +1806,37 @@ PSDKWrapper::subscribe_psdk_topics()
                    return_code);
     }
   }
+
+  if (params_.battery_level_frequency > 0)
+  {
+    return_code = DjiFcSubscription_SubscribeTopic(
+        DJI_FC_SUBSCRIPTION_TOPIC_BATTERY_SINGLE_INFO_INDEX1,
+        get_frequency(params_.battery_level_frequency), c_single_battery_index1_callback);
+
+    if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+    {
+      RCLCPP_ERROR(get_logger(),
+                   "Could not subscribe successfully to topic "
+                   "DJI_FC_SUBSCRIPTION_TOPIC_BATTERY_SINGLE_INFO_INDEX1, error %ld",
+                   return_code);
+    }
+  }
+
+  if (params_.battery_level_frequency > 0)
+  {
+    return_code = DjiFcSubscription_SubscribeTopic(
+        DJI_FC_SUBSCRIPTION_TOPIC_BATTERY_SINGLE_INFO_INDEX2,
+        get_frequency(params_.battery_level_frequency), c_single_battery_index2_callback);
+
+    if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+    {
+      RCLCPP_ERROR(get_logger(),
+                   "Could not subscribe successfully to topic "
+                   "DJI_FC_SUBSCRIPTION_TOPIC_BATTERY_SINGLE_INFO_INDEX2, error %ld",
+                   return_code);
+    }
+  }
+
```

Comment:
all this can be under the same `if (params_.battery_level_frequency > 0)` used for the battery topic

Reply:
already changed